src/io/flutter/run/daemon/FlutterAppManager.java

```diff
@@ -282,8 +291,52 @@ private void eventLogMessage(@NotNull LogMessage message, @NotNull FlutterDaemon
 
   private void eventLogMessage(@NotNull AppLog message, @NotNull FlutterDaemonController controller) {
     RunningFlutterApp app = findApp(controller, message.appId);
-    if (app != null && message.log != null) {
-      app.getConsole().print(message.log + "\n", ConsoleViewContentType.NORMAL_OUTPUT);
+    if (app != null) {
+      if (message.progress) {
+        if (message.finished) {
+          clearLatch();
+        } else {
+          // TODO: Is this the correct API to use to get the current project?
+          Project project = ProjectManager.getInstance().getDefaultProject();
```

Comment:
Hmmmm...  As per our conversation I don't think this is quite right.  Better probably to have the project passed down through the `FlutterDaemonController`.  Alternatively it could pass in the `projectDir` and you could look for a match in `ProjectManager.getInstance().getOpenProjects()`.